react/features/base/lib-jitsi-meet/functions.any.js

```diff
@@ -48,6 +48,32 @@ export function isAnalyticsEnabled(stateful: Function | Object) {
     return !(disableThirdPartyRequests || analytics.disabled);
 }
 
+/**
+ * Determines whether an avatar URL is valid.
+ *
+ * @param {Function|Object} stateful - The redux store, state, or
+ * {@code getState} function.
+ * @param {string} avatarURL - The avatar url to check.
+ * @returns {boolean} If the URL does not match the origin and disableThirdPartyRequests
+ * is defined, {@code false}; otherwise {@code true};
+ * {@code false}, otherwise.
+ */
+export function isAvatarUrlValid(stateful: Function | Object, avatarURL: string) {
+    const state = toState(stateful);
+    const { locationURL } = state['features/base/connection'];
+    const { disableThirdPartyRequests } = state['features/base/config'];
+
+    if (disableThirdPartyRequests) {
+        if (!locationURL || !locationURL.origin) {
+            return false;
+        }
+
+        return avatarURL && avatarURL.startsWith(locationURL.origin);
```

Comment:
This fails if I am in https://meet.jit.si and the avatar is in https://meet.jit.si.evil-domain.com

You need to parse the URL (just create a URL object out of the string) and compare origins.

Reply:
Thanks for the review and sorry for this mistake ! I'll fix it as soon as I can